### ISSUE TITLE
[Swift Implement new Concept Exercise: strings-and-characters

### ISSUE THREAD

**Issue by wneumann** [exercism/v3#2081]
This issue describes how to implement the `strings-and-characters` concept exercise for the Swift track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the String type [Swift][strings].

## Learning objectives

- Know of the existence of the `String` type.
- Know how to create a string.
- Know of the existence of `String` methods and properties
- Know some basic string methods (like finding the index of a character, or splitting a string based on a delimiter).
- Know how to combine strings.
- Know how to iterate over a string
- Know the existence of the `Character` type
- Know how the create a Character
- Know of the existence of `Character` methods and properties
- Know some basic `Character` methods
- Know how to append a `Character` onto a `String`
- Know how to convert a `Character` to a `String`

## Out of scope

- String indexing
- String decomposition
- String conformance to `Collection`.
- String slicing/Substrings.
- Multi-line strings.
- Raw strings.
- Using standard or custom format strings.
- Type inference.
- Memory and performance characteristics.
- Character views, utf8 views, utf16 views, etc.

## Concepts

- `strings`: Know of the existence of the `String` type; know how to create a string; know of the existence of `String` methods and properties; know some basic string methods; know how to combine strings; know how to iterate over a string
- `characters`: Know the existence of the `Character` type; know how the create a Character; know of the existence of `Character` methods and properties; know some basic `Character` methods; know how to append a `Character` onto a `String`; know how to convert a `Character` to a `String`

## Prequisites

- `basics`: know what a value is; know how to define a constant; know how to define a variable; know how to define a function

## Representer

TBD

## Analyzer

TBD

### Hints

- [Strings and Characters - Language Guide][strings]
- [`String` API documentation][string-docs]

### After

- [Strings and Characters - Language Guide][strings]

## Implementing

See [the Swift track implementation guide][swift-implementing-concept-exercises] for guidance.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[strings]: https://docs.swift.org/swift-book/LanguageGuide/StringsAndCharacters.html
[string-docs]: https://developer.apple.com/documentation/swift/string
[swift-implementing-concept-exercises]: https://github.com/wneumann/v3/blob/master/languages/swift/reference/implementing-a-concept-exercise.md

**Comment by wneumann** [comment]
Note that this is going to be the first 2/3 or so of the existing `strings` exercise with extra added in to cover `Characters`.

**Comment by haunshila** [comment]
I want to help, could you please assign this issue to me.

**Comment by wneumann** [comment]
Hello.

First of all, thank you for offering you time to help out with the project. I really appreciate it.

Regarding this particular exercise, I've already got it about 90% done and ready to submit. So if you want to help on this one (and the following `strings` exercise which is also mostly done) I can definitely tag you to help with the review. And if you are looking to start writing a concept exercise, we do have [quite a few open issues for them](https://github.com/exercism/v3/issues?q=is%3Aissue+is%3Aopen+%5BSwift%5D+Implement+new+Concept+Exercise%3A+). I would suggest [`bit-manipulation`](https://github.com/exercism/v3/issues/1666) or [`sets`](https://github.com/exercism/v3/issues/1669) for a good first task. Let me know if you'd like to work on either of those and I'll assign them to you.

I see you've joined the slack. Be sure to join the #track-swift and #maintaining-swift channels, perhaps they'll become active again now that I'm not the only one there. Let me know if you have any questions.

**Comment by haunshila** [comment]
Sure , I joined both the channel.
Please assign bit-manipulation as a starting point.

Thanks

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by wneumann** [comment]
OK, I went to add you and you aren't yet in the list of contributors to choose from. If you haven't yet, you may want to email contribute@exercism.io, let them know you would like to contribute, and they will help you get set up to contribute. In the meantime, you can start working on the the exercise. Be sure to read through the [Exercism contribution documents](https://github.com/exercism/v3/blob/master/docs/maintainers/README.md) and the [Swift track contribution guide](https://github.com/wneumann/v3/blob/master/languages/swift/reference/implementing-a-concept-exercise.md).

Let me know if you have any questions.

**Comment by wneumann** [comment]
Exercise added in PR https://github.com/exercism/v3/pull/2141

**Comment by wneumann** [comment]
PR https://github.com/exercism/v3/pull/2141 merged.
